controllers/k8ssandra/medusa_reconciler.go

```diff
@@ -44,9 +46,27 @@ func (r *K8ssandraClusterReconciler) reconcileMedusa(
 			return res
 		}
 
+		medusaContainer := medusa.CreateMedusaMainContainer(dcConfig, medusaSpec, kc.Spec.UseExternalSecrets(), kc.SanitizedName(), logger)
 		medusa.UpdateMedusaInitContainer(dcConfig, medusaSpec, kc.Spec.UseExternalSecrets(), kc.SanitizedName(), logger)
-		medusa.UpdateMedusaMainContainer(dcConfig, medusaSpec, kc.Spec.UseExternalSecrets(), kc.SanitizedName(), logger)
-		medusa.UpdateMedusaVolumes(dcConfig, medusaSpec, kc.SanitizedName())
+		medusa.UpdateMedusaMainContainer(dcConfig, medusaContainer)
+
+		// Create required volumes for the Medusa containers
+		volumes, additionalVolumes := medusa.GenerateMedusaVolumes(dcConfig, medusaSpec, kc.SanitizedName())
+		for _, volume := range volumes {
```

Comment:
Not a big fan of using both volumes and additionalVolumes. Can't we just use one or the other?

Reply:
I think we should get rid of the additionalVolumes now, because the testing strategy can't rely anymore on local volumes anyway.
It's the only reason why we were using them I think.
One thing I'm worried about is that it would break updates if someone is using the volumes 🤔 cass-operator won't allow changing the list of additional volumes on an existing cluster.

Reply:
One for us to discuss later this evening?

Reply:
Where did we land on this? Are we just going to add release notes with a migration pathway?